xc/common/cmake/vivado.cmake

```diff
@@ -464,3 +478,199 @@ function(ADD_VIVADO_PNR_TARGET)
       DEPS ${DEPS}
       BITSTREAM ${BITSTREAM})
 endfunction()
+
+function(PREPARE_RAPIDWRIGHT)
+  # Creates a target rapidwright with the following envirnoment variables:
+  #  RAPIDWRIGHT_INSTALLED - True if a valid RapidWright installation is
+  #                          found at the variable RAPIDWRIGHT_PATH.
+  #  RAPIDWRIGHT_PATH - If RAPIDWRIGHT_INSTALLED=TRUE, this points to the
+  #                     directory where RapidWright is located.
+  #  JAVA - If RAPIDWRIGHT_INSTALLED=TRUE, this points to the location of
+  #         the Java runtime executable needed invoke RapidWright entry
+  #         points.
+  #  INVOKE_RAPIDWRIGHT - If RAPIDWRIGHT_INSTALLED=TRUE, this points to the
+  #         location of the invoke_rapidwright.sh script which can be used to
+  #         invoke RapidWright.
+  add_custom_target(rapidwright)
+  set_target_properties(rapidwright PROPERTIES RAPIDWRIGHT_INSTALLED FALSE)
+
+  set(HAVE_RAPIDWRIGHT FALSE)
+  set(HAVE_JAVA FALSE)
+  set(HAVE_URAY_VIVADO FALSE)
+  if(EXISTS "${RAPIDWRIGHT_PATH}" AND EXISTS ${RAPIDWRIGHT_PATH}/interchange AND EXISTS ${RAPIDWRIGHT_PATH}/scripts/invoke_rapidwright.sh)
+    set(HAVE_RAPIDWRIGHT TRUE)
+  endif()
+
+  if(${HAVE_RAPIDWRIGHT})
+    find_program(JAVA java)
+    if(EXISTS ${JAVA})
+      set(HAVE_JAVA TRUE)
+    endif()
```

Comment:
Does java need to be pre-existing?

If so, wouldn't it be better (if doable) to have a Java package in conda, so that we can be sure the correct version is used?

Reply:
Because of make-env, java should be on the path, whether via conda or otherwise.  So from a CMake perspective, this is right.  Whether we should do an `apt-get install java` or `conda install java` is a different problem.